src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -0,0 +1,584 @@
+        if (err.code() == abs_error_code::_unknown) {
+            vlog(
+              abs_log.error,
+              "Accessing storage account {}, unexpected REST API error "
+              "detected: code={}, http_code={}; {}",
+              bucket,
+              err.code_string(),
+              err.http_code(),
+              err.message());
+            outcome = error_outcome::fail;
+        } else if (
+          err.http_code() == status::forbidden
+          || err.http_code() == status::unauthorized) {
+            vlog(
+              abs_log.error,
+              "Received {} error response from ABS. This indicates "
+              "misconfiguration of ABS and/or Redpanda: {}",
+              err.code_string(),
+              err.message());
+            outcome = error_outcome::fail;
+        } else if (
+          err.code() == abs_error_code::container_being_disabled
+          || err.code() == abs_error_code::container_being_deleted
+          || err.code() == abs_error_code::container_not_found) {
+            vlog(
+              abs_log.error,
+              "Received {} error response from ABS. This indicates "
+              "that your container is not available. Remediate the issue for "
+              "uploads to resume: {}",
+              err.code_string(),
+              err.message());
+            outcome = error_outcome::fail;
+        } else if (err.code() == abs_error_code::blob_not_found) {
+            // Unexpected 404s are logged by 'request_future' at warn
+            // level, so only log at debug level here.
+            vlog(abs_log.debug, "BlobNotFound response received {}", key);
+            outcome = error_outcome::key_not_found;
+        } else if (err.code() == abs_error_code::server_busy) {
+            vlog(
+              abs_log.warn,
+              "{} response received for container {}",
+              err.code_string(),
+              bucket);
+            outcome = error_outcome::retry_slowdown;
+        } else {
+            vlog(
+              abs_log.trace,
+              "Received {} retryable error resopnse from ABS: {}",
+              err.code_string(),
+              err.message());
+            outcome = error_outcome::retry;
+        }
+    } catch (...) {
+        _probe->register_failure(abs_error_code::_unknown);
+
+        outcome = util::handle_client_transport_error(
+          std::current_exception(), abs_log);
+    }
+
+    co_return outcome;
+}
+
+ss::future<result<http::client::response_stream_ref, error_outcome>>
+abs_client::get_object(
+  bucket_name const& name,
+  object_key const& key,
+  const ss::lowres_clock::duration& timeout,
+  bool expect_no_such_key) {
+    return send_request(
+      do_get_object(name, key, timeout, expect_no_such_key), name, key);
+}
+
+ss::future<http::client::response_stream_ref> abs_client::do_get_object(
+  bucket_name const& name,
+  object_key const& key,
+  const ss::lowres_clock::duration& timeout,
```

Comment:
this is an integer. no real need to pass by reference

Reply:
Done in [force-push](https://github.com/redpanda-data/redpanda/compare/b8dace186bd904cfb634d13675389bc514eb6759..56c2f4b09eb678e762a99057b07de7756ae8ee8d).